components/blitz/src/omero/gateway/facility/TablesFacility.java

```diff
@@ -0,0 +1,526 @@
+import omero.gateway.SecurityContext;
+import omero.gateway.exception.DSAccessException;
+import omero.gateway.exception.DSOutOfServiceException;
+import omero.gateway.model.AnnotationData;
+import omero.gateway.model.DataObject;
+import omero.gateway.model.FileAnnotationData;
+import omero.gateway.model.FileData;
+import omero.gateway.model.ImageData;
+import omero.gateway.model.MaskData;
+import omero.gateway.model.PlateData;
+import omero.gateway.model.ROIData;
+import omero.gateway.model.TableData;
+import omero.gateway.model.WellSampleData;
+import omero.grid.BoolColumn;
+import omero.grid.Column;
+import omero.grid.Data;
+import omero.grid.DoubleArrayColumn;
+import omero.grid.DoubleColumn;
+import omero.grid.FileColumn;
+import omero.grid.FloatArrayColumn;
+import omero.grid.ImageColumn;
+import omero.grid.LongArrayColumn;
+import omero.grid.LongColumn;
+import omero.grid.MaskColumn;
+import omero.grid.PlateColumn;
+import omero.grid.RoiColumn;
+import omero.grid.SharedResourcesPrx;
+import omero.grid.StringColumn;
+import omero.grid.TablePrx;
+import omero.grid.WellColumn;
+import omero.model.FileAnnotation;
+import omero.model.FileAnnotationI;
+import omero.model.Image;
+import omero.model.ImageI;
+import omero.model.OriginalFile;
+import omero.model.OriginalFileI;
+import omero.model.Plate;
+import omero.model.PlateI;
+import omero.model.Roi;
+import omero.model.RoiI;
+import omero.model.WellSample;
+import omero.model.WellSampleI;
+import edu.emory.mathcs.backport.java.util.Arrays;
+
+/**
+ * {@link Facility} to interact with OMERO.tables
+ * 
+ * @author Dominik Lindner &nbsp;&nbsp;&nbsp;&nbsp; <a
+ *         href="mailto:d.lindner@dundee.ac.uk">d.lindner@dundee.ac.uk</a>
+ */
+public class TablesFacility extends Facility {
+    
+    /**
+     * Creates a new instance
+     * 
+     * @param gateway
+     *            Reference to the {@link Gateway}
+     */
+    TablesFacility(Gateway gateway) {
+        super(gateway);
+    }
+
+    /**
+     * Adds a new table with the provided data
+     * 
+     * @param ctx
+     *            The {@link SecurityContext}
+     * @param target
+     *            The object to attach the table to
+     * @param name
+     *            A name for the table (can be <code>null</code>)
+     * @param data
+     *            The data
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public void addTable(SecurityContext ctx, DataObject target, String name,
+            TableData data) throws DSOutOfServiceException, DSAccessException {
+        try {
+            if (name == null)
+                name = UUID.randomUUID().toString();
+
+            String[] columnNames = data.getColumnNames() != null ? data
+                    .getColumnNames() : new String[0];
```

Comment:
https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/ArrayUtils.html#EMPTY_STRING_ARRAY should save you the new instance allocation. Again, more an FYI than a blocker.